.cicd/submodule-regression-check.sh

```diff
@@ -28,7 +33,17 @@ for k in "${!BASE_MAP[@]}"; do
     echo "  timestamp on $BASE_BRANCH: $base_ts"
     if (( $pr_ts < $base_ts)); then
         echo "$k is older on $CURRENT_BRANCH than $BASE_BRANCH; investigating..."
-        if for c in `git log $CURRENT_BRANCH ^$BASE_BRANCH --pretty=format:"%H"`; do git show --pretty="" --name-only $c; done | grep -q "^$k$"; then
+        if [[ $TRAVIS == true && ! -z $TRAVIS_PULL_REQUEST_SLUG ]]; then # IF it's a forked PR, we need to switch back to the PR ref/head so we can git log properly
+            echo "git fetch origin +refs/pull/$TRAVIS_PULL_REQUEST/merge:"
+            git fetch origin +refs/pull/$TRAVIS_PULL_REQUEST/merge: &> /dev/null
+            echo "switching back to $TRAVIS_PULL_REQUEST_SLUG:$TRAVIS_PULL_REQUEST_BRANCH ($TRAVIS_COMMIT)"
+            echo 'git checkout -qf FETCH_HEAD'
+            git checkout -qf FETCH_HEAD &> /dev/null
+        elif [[ $BUILDKITE == true ]]; then
+            echo "switching back to $CURRENT_BRANCH"
+            git checkout -f $CURRENT_BRANCH &> /dev/null
```

Comment:
I don't see why this is needed.

Reply:
Nevermind. It seems to be needed to handle forked PR builds in Buildkite properly.